src/provisioning-ofono.c

```diff
@@ -1230,10 +1278,9 @@ int provisioning_init_ofono(void)
  * Currently this assumes that only one modem in ofono if need to support more
  * it could be something like below
  */
-#ifdef MULTIMODEM
 	modem_list = g_hash_table_new_full(g_str_hash, g_str_equal,
-						NULL, remove_modem);
-#endif
+						NULL, remove_modems);
+
```

Comment:
It will be called more that once, and even recursively, like this:
```
(gdb) bt
#0  remove_modems () at src/provisioning-ofono.c:1079
#1  0xb6f17eac in g_hash_table_remove_all_nodes (notify=1, hash_table=0x20c20) at ghash.c:500
#2  g_hash_table_remove_all (hash_table=0x20c20) at ghash.c:1347
#3  0x0000d6dc in remove_modems () at src/provisioning-ofono.c:1102
#4  0xb6f17eac in g_hash_table_remove_all_nodes (notify=1, hash_table=0x20c20) at ghash.c:500
#5  g_hash_table_remove_all (hash_table=0x20c20) at ghash.c:1347
#6  0x0000d6dc in remove_modems () at src/provisioning-ofono.c:1102
#7  0x0000af24 in clean_provisioning () at src/provisioning-ofono.c:88
#8  0x0000afb8 in check_idle () at src/provisioning-ofono.c:106
#9  0xb6f28b86 in g_main_dispatch (context=0x1c180) at gmain.c:3066
...
```
This doesn't make sense to me.